### ISSUE TITLE
Problems identifying attachments Office365

### ISSUE THREAD

**Issue by eduardom456** [Webklex/php-imap#370]
**Code to Reproduce**
```php
foreach($messages as $message) {
      $counter = 0;
      echo "Counter 01 = " . $message->getAttachments()->count();
      $aAttachment = $message->getAttachments();
      echo "Counter 02 = " . count($aAttachment);
      foreach($aAttachment as $oAttachment) {
            $counter++;
            echo "<p>".$val." ".$oAttachment->name."</p>";
      }
      echo "Counter 03 = " . $counter;
}
```

**Expected behavior**
Correct identification of the number of attachments contained in the received email.

**Desktop / Server:**
- PHP: 7.4.28
- Version v2.3.1
- Provider Outlook

**Additional context**

I have a problem with looping attachments.

For example, I receive an email with 15 attachments, but in processing the function identifies 14.

I have to reprocess the email to correct one or more missing attachments, and even then the problem can still persist, find 10, 13.

The above code was just the "extra" validations I performed to check the number of attachments.
basically, all counters match, however all conditions are required in what the function initially identified, i.e. if identified 14, counters will play 14, etc.

Thanks.

**Comment by eduardom456** [comment]
The above code was just the "extra" validations I performed to check the number of attachments.

Basically, all counters match, however all conditions are required in what the function initially identified, i.e. if identified 14, counters will play 14, etc.

**Comment by AdrianKuriata** [comment]
I think there is a problem with pushing attachments to the collection, here is a solution. Try it and check if this help you.

https://github.com/Webklex/php-imap/pull/372

**Comment by Webklex** [comment]
Hi @eduardom456 ,
many thanks for your well written report. I very much appreciate it!

Please update to the latest release and give it another try. There has been a ton of changes :)

Once again, thanks for taking the time and effort to make this library better!

Best regards & happy coding,
